packages/test/test-end-to-end-tests/src/test/benchmark/opCriticalPath.time.spec.ts

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { ITestDataObject, describeCompat } from "@fluid-private/test-version-utils";
+import { benchmark } from "@fluid-tools/benchmark";
+import { IContainer } from "@fluidframework/container-definitions/internal";
+import {
+	ContainerRuntime,
+	DefaultSummaryConfiguration,
+} from "@fluidframework/container-runtime/internal";
+import {
+	ITestContainerConfig,
+	ITestObjectProvider,
+	timeoutPromise,
+} from "@fluidframework/test-utils/internal";
+
+const testContainerConfig: ITestContainerConfig = {
+	runtimeOptions: {
+		enableGroupedBatching: true,
+		summaryOptions: {
+			initialSummarizerDelayMs: 0, // back-compat - Old runtime takes 5 seconds to start summarizer without thi
+			summaryConfigOverrides: {
+				...DefaultSummaryConfiguration,
+				...{ maxOps: 10, initialSummarizerDelayMs: 0, minIdleTime: 10, maxIdleTime: 10 },
+			},
+		},
+	},
+};
+
+type Patch<T, U> = Omit<T, keyof U> & U;
+
+type ContainerRuntime_WithPrivates = Patch<ContainerRuntime, { flush: () => void }>;
+
+describeCompat(
+	"Op Critical Paths - runtime benchmarks",
+	"NoCompat",
+	(getTestObjectProvider) => {
+		let provider: ITestObjectProvider;
+		let mainContainer: IContainer;
+		let defaultDataStore: ITestDataObject;
+		let containerRuntime: ContainerRuntime_WithPrivates;
+
+		const before = async () => {
+			provider = getTestObjectProvider();
+			const loader = provider.makeTestLoader(testContainerConfig);
+			mainContainer = await loader.createDetachedContainer(provider.defaultCodeDetails);
+
+			await mainContainer.attach(provider.driver.createCreateNewRequest());
+			defaultDataStore = (await mainContainer.getEntryPoint()) as ITestDataObject;
+			containerRuntime = defaultDataStore._context
+				.containerRuntime as ContainerRuntime_WithPrivates;
+
+			defaultDataStore._root.set("force", "write connection");
+			await provider.ensureSynchronized();
+		};
+		function sendOps(label: string) {
+			Array.from({ length: 100 }).forEach((_, i) => {
+				defaultDataStore._root.set(`key-${i}`, `value-${label}`);
+			});
+
+			containerRuntime.flush();
+		}
+
+		benchmark({
+			title: "Submit+Flush",
+			before, // Set up container for a write connection
+			benchmarkFnAsync: async () => {
+				sendOps("A");
+				const opsSent = await timeoutPromise<number>(
+					(resolve) => {
+						containerRuntime.deltaManager.outbound.once("idle", resolve);
+					},
+					{ errorMsg: "container2 outbound queue never reached idle state" },
+				);
+				assert(opsSent > 0, "Expecting op(s) to be sent.");
+			},
+		});
+	},
+);
+
+describeCompat(
```

Comment:
I couldn't get it to work putting both benchmarks under the same describe so I just copied all the code.

Reply:
That's weird, curious about what was going wrong because it should definitely be possible to have several benchmarks in one describe block.

Reply:
This is a real shame :confused: Probably something to be looked into by the owner of the benchmark package

Reply:
Let's see the error first, because we do have places with more than one benchmark inside one describe that are working fine.

Reply:
I will try again this morning.  I was kind of tired while working on this, I'll admit 🥱

Reply:
It's something about the container state when they both are running.  I'm opening a follow-up task about it.  If I put them together as-is, the container closes at some point during the 2nd benchmark due to error:  `No context for op`

``` 1) Op Critical Paths - runtime benchmarks
       Non-Compat
         "after each" hook for "@Benchmark @Measurement @ExecutionTime Roundtrip":
     Error: Unexpected Errors in Logs:
[
  {
    "category": "error",
    "eventName": "fluid:telemetry:Container:ContainerClose",
    "error": "No context for op",
    "errorType": "dataProcessingError",
    "...": "*** Additional properties not shown, see full log for details ***"
  }
]
```

Reply:
AB#19812